include/exadg/operators/generalized_laplace_operator.h

```diff
@@ -0,0 +1,781 @@
+      flags.boundary_faces = dealii::update_JxW_values | dealii::update_gradients |
+                             dealii::update_normal_vectors | dealii::update_quadrature_points;
+
+    return flags;
+  }
+
+  /**
+   * Calculates and returns the `flux` inside the cell integral. This flux should be weighted with
+   * the gradient of the shape functions.
+   */
+  static inline DEAL_II_ALWAYS_INLINE //
+    GradientType
+    get_volume_flux(GradientType const & gradient, CoefficientType const & coefficient)
+  {
+    return coeff_mult(coefficient, gradient);
+  }
+
+  /**
+   * Calculates and returns the `flux` on the faces which should be weighted with the gradient of
+   * the shape functions.
+   */
+  static inline DEAL_II_ALWAYS_INLINE //
+    GradientType
+    calculate_gradient_flux(ValueType const &       value_m,
+                            ValueType const &       value_p,
+                            Vector const &          normal,
+                            CoefficientType const & coefficient)
+  {
+    ValueType const    jump_value  = value_m - value_p;
+    GradientType const jump_tensor = outer_product(jump_value, normal);
+
+    return -0.5 * coeff_mult(coefficient, jump_tensor);
+  }
+
+  /**
+   * Calculates and returns the `flux` on the faces which should be weighted with the normal
+   * derivative of the shape functions.
+   *
+   * Weighting with the normal derivative (gradient times face normal) is mathematically only
+   * possible if the coefficient is non-coupling type. So, this is asserted inside the function
+   * body.
+   */
+  static inline DEAL_II_ALWAYS_INLINE //
+    ValueType
+    calculate_normal_derivative_flux(ValueType const &       value_m,
+                                     ValueType const &       value_p,
+                                     CoefficientType const & coefficient)
+  {
+    {Assert(coefficient_is_scalar,
+            dealii::ExcMessage("Normal derivative flux only makes"
+                               "sense with scalar coefficients."))}
+
+    ValueType const jump_value = value_m - value_p;
+
+    return -0.5 * coefficient * jump_value;
+  }
+
+  /**
+   * Calculates and returns the `flux` on the faces which should be weighted with the shape
+   * function values.
+   */
+  inline DEAL_II_ALWAYS_INLINE //
+    ValueType
+    calculate_value_flux(GradientType const &    gradient_m,
+                         GradientType const &    gradient_p,
+                         ValueType const &       value_m,
+                         ValueType const &       value_p,
+                         Vector const &          normal,
+                         CoefficientType const & coefficient)
+  {
+    ValueType const    jump_value  = value_m - value_p;
+    GradientType const jump_tensor = outer_product(jump_value, normal);
+
+    GradientType const average_gradient = 0.5 * (gradient_m + gradient_p);
+
+    return -coeff_mult(coefficient, (average_gradient - tau * jump_tensor)) * normal;
```

Comment:
That's because I didn't want to implement operator as the negative Laplace operator. But of course I can change it to be the negative Laplace as in other parts of ExaDG.